src/main/java/com/infrastructure/PostgressImpl.java

```diff
@@ -0,0 +1,36 @@
+package com.infrastructure;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.stereotype.Repository;
+
+import com.javapractice.gateways.ITodoRepository;
+
+@Repository
+public class PostgressImpl implements ITodoRepository {
+
+    private final String url = "jdbc:postgresql://localhost:5432/mydb";
+    private final String user = "username";
+    private final String password = "password";
+
```

Comment:
I missed the constructor that connects to the database: 
    public PostgressImpl() {
        try {
            conn = DriverManager.getConnection(url, user, password);
            System.out.println("Connected to the PostgreSQL server successfully.");
        } catch (Exception e) {
            e.printStackTrace();
        }
    }